fedimint-client/src/db.rs

```diff
@@ -418,8 +418,8 @@ pub fn get_core_client_database_migrations() -> BTreeMap<DatabaseVersion, CoreMi
                 };
 
                 let config_key = ClientConfigKey { id: id.id };
-                dbtx.insert_new_entry(&config_key, &config).await;
                 dbtx.remove_entry(&id).await;
+                dbtx.insert_new_entry(&config_key, &config).await;
```

Comment:
We should assert that we only found 0 or 1 existing configs, in case something ever changes in the future (we store more than one?) and this migration needs a fix.